src/router_core/router_core.c

```diff
@@ -258,6 +260,22 @@ void qdr_core_free(qdr_core_t *core)
         link = DEQ_HEAD(core->open_links);
     }
 
+    //
+    // Clean up any qdr_delivery_cleanup_t's that are still left in the core->delivery_cleanup_list
+    //
+    if (DEQ_SIZE(core->delivery_cleanup_list) > 0) {
```

Comment:
I'm confused - this check shouldn't be necessary since if length is zero the loop below will not execute since DEQ_HEAD/cleanup will == 0?

Reply:
Also why not simply call qdr_do_message_to_addr_free() here instead of duplicating the code?